sdk/core/azure-core/src/main/java/com/azure/core/util/polling/ChainedPollingStrategy.java

```diff
@@ -0,0 +1,101 @@
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A polling strategy that chains multiple polling strategies, finds the first strategy that can poll the current
+ * long running operation, and polls with that strategy.
+ *
+ * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+ *            should be kept
+ * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+ *            body should be kept
+ */
+public class ChainedPollingStrategy<T, U> implements PollingStrategy<T, U> {
+    private final List<PollingStrategy<T, U>> pollingStrategies;
+    private PollingStrategy<T, U> pollableStrategy = null;
+
+    /**
+     * Creates an empty chained polling strategy.
+     */
+    public ChainedPollingStrategy() {
+        this.pollingStrategies = new ArrayList<>();
+    }
+
+    /**
+     * Creates a chained polling strategy with 3 known polling strategies, {@link OperationResourcePollingStrategy},
+     * {@link LocationPollingStrategy}, and {@link StatusCheckPollingStrategy}, in this order. The first strategy that
+     * can poll on the initial response will be used. The created chained polling strategy is capable of handling most
+     * of the polling scenarios in Azure.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+     *            should be kept
+     * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+     *            body should be kept
+     * @return the initialized chained polling strategy with the default chain
+     */
+    public static <T, U> ChainedPollingStrategy<T, U> createDefault(
+            HttpPipeline httpPipeline,
+            Context context) {
+        return new ChainedPollingStrategy<T, U>()
+            .addPollingStrategy(new OperationResourcePollingStrategy<>(httpPipeline, context))
+            .addPollingStrategy(new LocationPollingStrategy<>(httpPipeline, context))
+            .addPollingStrategy(new StatusCheckPollingStrategy<>());
+    }
+
+    /**
+     * Adds a polling strategy to the chain of polling strategies.
+     * @param pollingStrategy the polling strategy to add
+     * @return the modified ChainedPollingStrategy instance
+     */
+    public ChainedPollingStrategy<T, U> addPollingStrategy(PollingStrategy<T, U> pollingStrategy) {
+        this.pollingStrategies.add(pollingStrategy);
+        return this;
+    }
```

Comment:
Do we want to make the polling strategy immutable once constructed? Maybe we can have the ctor take a list of polling strategies rather than having this make the strategy mutable. This will also allow us to have a no-args ctor that creates the default strategy. That way, we'll not have a static factory method & ctor - 2 different ways to create an instance.

Reply:
```java
public ChainedPollingStrategy(PollingStrategy... pollingStrategies) {
  // create chain with specified strategies 
}

public ChainedPollingStrategy() {
  // create empty chain or default chain?
}
```

Would it be confusing to create a chain with 3 items with the empty constructor?

Reply:
Another approach similar to how azure-identity does it is having another "default" type like `DefaultAzureCredential` in azure-identity.

```java
public class ChainedPollingStrategy {
  public ChainedPollingStrategy(PollingStrategy... pollingStrategies) {
    // create chain with specified strategies 
  }

  public ChainedPollingStrategy() {
    // create empty chain
  }
}

public class DefaultPollingStrategy {
  private ChainedPollingStrategy chainedPollingStrategy;

  public DefaultPollingStrategy(HttpPipeline httpPipeline, Context context) {
    this.chainedPollingStrategy = new ChainedPollingStrategy(
      new OperationResourcePollingStrategy(httpPipeline, context),
      new LocationPollingStrategy(httpPipeline, context),
      new StatusCheckPollingStrategy());
  }
}
```

Reply:
Updated with this design:

```java
public class ChainedPollingStrategy<T,U> {
  public ChainedPollingStrategy(List<PollingStrategy<T,U>> pollingStrategies) {
    // create chain with specified strategies 
  }
}

public class DefaultPollingStrategy {
  private ChainedPollingStrategy chainedPollingStrategy;

  public DefaultPollingStrategy(HttpPipeline httpPipeline, Context context) {
    this.chainedPollingStrategy = new ChainedPollingStrategy(
      new OperationResourcePollingStrategy(httpPipeline, context),
      new LocationPollingStrategy(httpPipeline, context),
      new StatusCheckPollingStrategy());
  }
}
```

Since varags don't work well with generic params.